ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/transactional/TransactionalInterceptorBase.java

```diff
@@ -97,18 +120,55 @@ private Transactional getTransactional(InvocationContext ic) {
             return transactional;
         }
 
-        // see if the target is a stereotype
-        for (Annotation annotation : interceptedBean.getBeanClass().getAnnotations()) {
+        // check existence of the stereotype on method
+        Transactional transactionalMethod = getTransactionalAnnotationRecursive(ic.getMethod().getAnnotations());
+        if(transactionalMethod != null) return transactionalMethod;
+        // stereotype added by an extension
+        AnnotatedType<?> annTypeBean = extension.getBeanToAnnotatedTypeMapping().get(interceptedBean);
+        Transactional transactionalExtension = getTransactionalAnnotationRecursive(annTypeBean.getAnnotations());
+        if(transactionalExtension != null) return transactionalExtension;
+        // stereotype taken from intercepted bean getStereotype
+        Transactional transactionalBasedOnStereotype = null;
+        for(Class<? extends Annotation> stereotype: interceptedBean.getStereotypes()) {
+            transactionalBasedOnStereotype = stereotype.getAnnotation(Transactional.class);
+            if(transactionalBasedOnStereotype != null) return transactionalBasedOnStereotype;
+        }
+        // stereotype taken from intercepted bean getStereotypeDefinition
```

Comment:
What's the use of this part of code? I think it basically duplicates job done in the above part. It should be sufficient to go through the `AnnotatedType`.

The only case I can think of is when the actual class is annotated by `@Transactional` but you deliberately remove this annotation from `AnnotatedType` - however, in this case I am not sure you actually want to intercept it. I'd wager you don't.